src/main/java/com/atherys/towns/facade/PlotSelectionFacade.java

```diff
@@ -105,50 +100,34 @@ private void sendPointSelectionMessage(Player player, String point) {
 
     /**
      * Validate a plot selection.
-     *
-     * @param selection the selection to be validated
-     * @throws CommandException if the plot selection is null, is incomplete ( either point A or point B is null ),
-     *                          it's area is greater than the maximum configured, or it's smallest side is smaller than the minimum configured
+     * @param player The player requesting the selection
      */
-    public boolean validatePlotSelection(PlotSelection selection, Player player, boolean messageUser, Location<World> location) {
-
+    public void validatePlayerPlotSelection(Player player) throws TownsCommandException {
+        PlotSelection selection = getCurrentPlotSelection(player);
         if (selection == null) {
-            if (messageUser) townMsg.error(player, "Plot selection is null.");
-            return false;
+            throw new TownsCommandException("Plot selection is null");
         }
 
         if (!selection.isComplete()) {
-            if (messageUser) townMsg.error(player, "Plot selection is incomplete.");
-            return false;
-        }
-
-        int selectionArea = getPlotSelectionArea(selection);
-        if (selectionArea > config.TOWN.MAX_PLOT_AREA) {
-            if (messageUser)
-                townMsg.error(player, "Plot selection has an area greater than permitted ( ", selectionArea, " > ", config.TOWN.MAX_PLOT_AREA, " )");
-            return false;
-        }
-
-        int smallestSide = getSmallestPlotSelectionSideSize(selection);
-        if (smallestSide < config.TOWN.MIN_PLOT_SIDE - 1) {
-            if (messageUser)
-                townMsg.error(player, "Plot selection has a side smaller than permitted ( ", smallestSide, " < ", config.TOWN.MIN_PLOT_SIDE, " )");
-            return false;
-        }
-
-        Plot plot = plotService.createPlotFromSelection(selection);
-        if (plotService.plotIntersectsAnyOthers(plot)) {
-            if (messageUser) townMsg.error(player, "The plot selection intersects with an already-existing plot.");
-            return false;
+            throw new TownsCommandException("Plot selection is incomplete.");
         }
+    }
 
-        if (!plotService.isLocationWithinPlot(location, plot)) {
-            if (messageUser) townMsg.error(player, "You must be within your plot selection!");
+    public boolean playerHasValidSelection(Player player) {
+        try {
+            validatePlayerPlotSelection(player);
+        } catch (TownsCommandException e) {
             return false;
         }
         return true;
     }
 
+    public PlotSelection getValidPlayerPlotSelection(Player player) throws TownsCommandException {
+        PlotSelection selection = getOrCreateSelection(player);
+        validatePlayerPlotSelection(player);
+        return selection;
+    }
+
```

Comment:
I understand why this was done, because both types of plots would have different validation. But I think the instant feedback feature is worth it to keep. Rynelf suggested possibly having plot selection default to checking town validation rules, unless a flag is passed in for a nation type.

Reply:
This most likely only needs to be moved back for borders right?
I'll probably keep them separated and just make borders call Plot + Town Validation by default?

Reply:
Yeah that works fine, I agree, it makes more sense to keep them separate.